server/controller/questions.js

```diff
@@ -17,13 +33,30 @@ exports.get = async (request, response) => {
 };
 
 exports.post = async (request, response) => {
+  const userId = request.id;
+  const dialectResult = await users.findOne({
+    raw: true,
+    where: { id: userId },
+    attributes: ['dialect_id'],
+  });
+  const { dialect_id: dialectId } = dialectResult;
   try {
     const owner = request.id;
     const { speclalizationsId, question, section } = request.body;
     const data = { owner, question, speclalization_id: speclalizationsId };
     await questions.create(data);
     const results = await getQuestions(section);
-    response.status(200).send(results[0]);
+    const getTranslationCount = results[0].map(async (element) => {
```

Comment:
@lubnaabd `getTranslationCount` is the same function here and up above, in exports.get (line 16).  If you have time, it might be better to refactor this to move getTranslationCount out of `get` and `post` and just have it at the top level of this file, so that get and post can both call the same function.  It's better to not have the function duplicated, because if you had to fix a bug in the function, you would have to fix it in both places (and you might forget one).

Reply:
If you did that, you might even be able to take it a step further and move the `const dialectResult = await users.findOne({` inside that new getTranslationCount function too.